docs/regression.ipynb

```diff
@@ -0,0 +1,696 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "x = data[:, 0]\n",
+    "y = data[:, 1]\n",
+    "\n",
+    "m, b = coef\n",
+    "line_x = np.linspace(np.min(x), np.max(x), 100)\n",
+    "line_y = m * line_x + b\n",
+    "\n",
+    "plt.scatter(x, y, color='blue', label='Data Points')\n",
+    "plt.plot(line_x, line_y, color='red', label=f'Line: y = {m}x + {b}')\n",
+    "\n",
+    "plt.xlabel('X-axis')\n",
+    "plt.ylabel('Y-axis')\n",
+    "plt.title('Scatter Plot with DP-Theil-Sen Regression')\n",
+    "plt.legend()\n",
+    "\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The DP regression results align well with our synthetic data. Now let's apply the method to some real data. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Performing a Differentially Private Regression on Real Data (Find Better Data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's apply this mechanism to real data! We will use the same Labor Force Survey data from France that we've been working with in other examples. In an earlier example #TODO (link exact one with line graph) we saw that there was a strong positive trend between year and the amount of people not working due to a lay off. Let's see if that relationship is linear and and apply the DP Theil Sen regression methods. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 116,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/var/folders/dg/nlwnz_jj7rsblr2tzwj962b40000gn/T/ipykernel_93345/4255135836.py:23: DeprecationWarning: `groupby` is deprecated. It has been renamed to `group_by`.\n",
+      "  year_actual = df.filter((pl.col(\"ILOSTAT\")==3)).groupby(\"YEAR\").agg(\n"
+     ]
+    }
+   ],
+   "source": [
+    "df = pl.scan_csv(\"sample_FR_LFS.csv\", infer_schema_length=1000, ignore_errors=True)\n",
+    "\n",
+    "year_actual = df.filter((pl.col(\"ILOSTAT\")==3)).groupby(\"YEAR\").agg(\n",
+    "    pl.col(\"YEAR\").count().alias(\"count\").cast(pl.Float64)\n",
+    ").sort(\"YEAR\").collect()"
```

Comment:
I worry that it confuses things to have non-DP use of polars. If you keep it, should be very clear about what's going on.